openpype/hosts/houdini/api/lib.py

```diff
@@ -649,3 +649,66 @@ def get_color_management_preferences():
         "display": hou.Color.ocio_defaultDisplay(),
         "view": hou.Color.ocio_defaultView()
     }
+
+
+def publisher_show_and_publish():
+    """Open publisher window and trigger publishing action. """
+
+    from openpype.tools.utils.host_tools import get_tool_by_name
+
+    main_window = get_main_window()
+    publisher_window = get_tool_by_name(
+        tool_name="publisher",
+        parent=main_window
+    )
+
+    publisher_window.set_current_tab("publish")
+    publisher_window.make_sure_is_visible()
+    publisher_window._reset_on_show = False
+
+    publisher_window._controller.reset()
+    publisher_window._controller.publish()
+
+
+def self_publish():
+    """Self publish from ROP nodes. """
+    from openpype.pipeline import registered_host
+    from openpype.pipeline.create import CreateContext
```

Comment:
Why are these imports here and not at the top?

Reply:
No reason actually.